kale/embed/gcn.py

```diff
@@ -0,0 +1,259 @@
+import numpy as np
+import torch
+from torch_geometric.utils import add_remaining_self_loops
+from torch_geometric.nn.conv import MessagePassing
+from torch_scatter import scatter_add
+from torch.nn import Parameter
+
+
+# Copy-paste with slight modification from torch_geometric.nn.GCNConv
+class GCNEncoderLayer(MessagePassing):
+    r"""
+    Modification of PyTorch Geometirc's nn.GCNConv, which reduces the computational cost of GCN layer for
+    `GripNet <https://github.com/NYXFLOWER/GripNet>`_ model.
+    The graph convolutional operator from the `"Semi-supervised Classification with Graph Convolutional Networks"
+    <https://arxiv.org/abs/1609.02907>`_ paper.
```

Comment:
Proper citation please (ICLR2017). Use arxiv only for preprint not published yet.